### DISCUSSION TITLE
Building a spectral BSDF model

### DISCUSSION THREAD

**Discussion by SylvainDoute** [eradiate/eradiate#244] [Q&A] [answered]
Hello Eradiate,
I am building a Martian 1D scene. I have already a working aerosol model and an adapted instrument response function (TGO/ExoMars CaSSIS, ESA). Now I need a realistic spectral BSDF model. I inverted the RPV model on some reflectance measurements I have for Mars. I am wondering how can I integrate the three experimental spectra into the surface_bsdf object 
surface_bsdf = ertsc.bsdfs.RPVBSDF(rho_0=spectrum1, rho_c=None, k=spectrum2, g= spectrum3)
spectrum1, spectrum2, spectrum3 should be of type spectrum but I do not know how to create them and fill them with my data. I went through the documentation but I cannot figure it out.
Thanks in advance for your help.
Mes best regards,
Sylvain Douté
![image](https://user-images.githubusercontent.com/77278793/175018880-97449218-a37d-4548-a316-911262af8633.png)

**Comment by leroyvn** [comment, marked as answer]
Hi @SylvainDoute, I think what you are looking for is the [`InterpolatedSpectrum`](https://eradiate.readthedocs.io/en/stable/rst/reference_api/generated/autosummary/eradiate.scenes.spectra.InterpolatedSpectrum.html) type. You can either instantiate it directly, or use its directionary specification:
```python
spectrum1 = {"type": "interpolated", "values": ..., "wavelengths": ...}
spectrum2 = InterpolatedSpectrum(values=..., wavelengths=...)
 ```
Note that the upcoming v0.22.5, which we intend to release by the end of the Summer, will also have a NetCDF loader for these spectra.

**Reply by SylvainDoute** [reply]
thank you Vincent. That was simple. Myself I wrote a small json loader for my spectra.
If I may ask another question. Upon running the experiment I end up with a long log full of calls to my aerosol model netcdf file. Could I get rid of all these lines ? 
![image](https://user-images.githubusercontent.com/77278793/175062605-8bab1dfe-8bb2-4dd7-b74b-b247e3685fbe.png)

**Reply by leroyvn** [reply]
Yes: you can redirect the logs to the Python logging framework and they will be hidden by default. To do so, you have to invoke
```python
eradiate.kernel.logging.install_logging()
```
Note that this is the most minimal thing to do. I actually recommend to also activate notebook support:
```python
eradiate.notebook.install()
```
or with the notebook magic
```
%load_ext Eradiate
```

And with v0.22.5, the IPython extension will do all that automatically.

**Reply by leroyvn** [reply]
Sorry, I misunderstood: I think what you're seeing is a bug, I must have left a print somewhere in the NetCDF loader... I'll look it up.

**Reply by leroyvn** [reply]
Can you tell me if this is a particle property or an atmospheric property dataset?

**Reply by leroyvn** [reply]
Can you tell me which version of Eradiate you are running? I cannot find any print statement in the current code base.
